include/envoy/router/router.h

```diff
@@ -227,6 +239,12 @@ class RetryPolicy {
    * @return absl::optional<std::chrono::milliseconds> maximum retry interval
    */
   virtual absl::optional<std::chrono::milliseconds> maxInterval() const PURE;
+
+  /**
+   * @return absl::optional<RetryBudget> limit on allowed concurrent retries in relation to current
+   * outstanding requests.
+   */
+  virtual absl::optional<RetryBudget> retryBudget() const PURE;
```

Comment:
doesnt really matter much here since RetryBudget is small, but if you return by const ref here you avoid a copy